### PR TITLE
Red pjama 3b instruct

### PR DESCRIPTION
**Pull request by akiseakusa**
This should close https://github.com/openvinotoolkit/openvino.genai/issues/276 , I have updated the model list , and added tests taking reference from issue description at https://github.com/openvinotoolkit/openvino.genai/issues/259 , @p-wysocki @ilya-lavrenov @pavel-esir in the readme I have added Red pjama 3b separately along with that huggingface comparison has also added. please let me know if anything else can be done.

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
Please, resolve merge conflicts.

**Comment by ilya-lavrenov** [comment]
Please, resolve merge conflicts

**Comment by akiseakusa** [comment]
@ilya-lavrenov Sorry for the delay,I have resolved the conflict. please check.

File: .github/workflows/causal_lm_cpp.yml [hunk 1]
```diff
@@ -12,6 +12,87 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  cpp-beam_search_causal_lm-red-pajama-3b-instruct:
+    runs-on: ubuntu-20.04-16-cores
```
Comment (Wovchena) [inline .github/workflows/causal_lm_cpp.yml:16]:
```suggestion
    runs-on: ubuntu-20.04
```
Can you try the smaller runner?

File: .github/workflows/causal_lm_cpp.yml [hunk 2]
```diff
@@ -12,6 +12,87 @@ concurrency:
...
+        run: |
+          source ./ov/setupvars.sh
+          python3 -m pip install --upgrade-strategy eager "optimum>=1.14" -r ./llm_bench/python/requirements.txt ./thirdparty/openvino_tokenizers/[transformers] --extra-index-url https://download.pytorch.org/whl/cpu && python3 ./llm_bench/python/convert.py --model_id togethercomputer/RedPajama-INCITE-Instruct-3B-v1 --output_dir ./RedPajama-INCITE-Instruct-3B-v1/ --precision FP16
+          cmake -DCMAKE_BUILD_TYPE=Release -S ./text_generation/causal_lm/cpp/ -B ./build/
+          cmake --build ./build/ --config Release -j8
+      - name: Compare
+        run: |
+          source ./ov/setupvars.sh
+          convert_tokenizer ./RedPajama-INCITE-Instruct-3B-v1/pytorch/dldt/FP16/ --output ./RedPajama-INCITE-Instruct-3B-v1/pytorch/dldt/FP16/ --with-detokenizer --trust-remote-code
+          timeout 50s ./build/beam_search_causal_lm .RedPajama-INCITE-Instruct-3B-v1/pytorch/dldt/FP16/ 69 > ./pred.txt
```
Comment (Wovchena) [inline .github/workflows/causal_lm_cpp.yml:39]:
```suggestion
          timeout 50s ./build/beam_search_causal_lm ./RedPajama-INCITE-Instruct-3B-v1/pytorch/dldt/FP16/ 69 > ./pred.txt
```

File: .github/workflows/causal_lm_cpp.yml [hunk 3]
```diff
@@ -12,6 +12,87 @@ concurrency:
...
+          for beam in transformers.LlamaForCausalLM.from_pretrained('togethercomputer/RedPajama-INCITE-Instruct-3B-v1').generate(**tokenized, num_beam_groups=3, num_beams=15, num_return_sequences=15, diversity_penalty=1.0, max_length=20, early_stopping=False, length_penalty=1.0, no_repeat_ngram_size=9, do_sample=False):
+              ref = ': ' + tokenizer.decode(beam[tokenized['input_ids'].numel():], skip_special_tokens=True) + '\n'
+              idx = predictions.find(ref)
+              if -1 == idx:
+                  raise RuntimeError(f'Missing "{ref=}" from predictions')
+              predictions = predictions[:idx] + predictions[idx + len(ref):]
+          "
+          echo 69 passed
+
+  cpp-beam_search_causal_lm-Mistral-7B:
```
Comment (Wovchena) [inline .github/workflows/causal_lm_cpp.yml:55]: I believe it won't ever pass, please remove it. https://github.com/openvinotoolkit/openvino.genai/pull/371

File: .github/workflows/causal_lm_cpp.yml [hunk 4]
```diff
@@ -396,46 +475,3 @@ jobs:
               predictions = predictions[:idx] + predictions[idx + len(ref):]
           "
           echo Phi-1_5 passed
-          
-  cpp-greedy_causal_lm-redpajama-3b-chat:
```
Comment (Wovchena) [inline .github/workflows/causal_lm_cpp.yml:400]: Why do you remove it? Can you merge it with your test implementation

File: text_generation/causal_lm/cpp/README.md [hunk 5]
```diff
@@ -148,7 +148,6 @@ To enable Unicode characters for Windows cmd open `Region` settings from `Contro
    2. https://huggingface.co/microsoft/phi-1_5
 9. [notus-7b-v1](https://huggingface.co/argilla/notus-7b-v1)
 10. [zephyr-7b-beta](https://huggingface.co/HuggingFaceH4/zephyr-7b-beta)
-11. [redpajama-3b-chat](https://huggingface.co/ikala/redpajama-3b-chat)
```
Comment (Wovchena) [inline text_generation/causal_lm/cpp/README.md:151]: Why do you remove [redpajama-3b-chat](https://huggingface.co/ikala/redpajama-3b-chat)?

**Review by Wovchena — CHANGES_REQUESTED** [review]

**Comment by andrei-kochin** [comment]
@akiseakusa could you please apply the comments?

**Pull request closed without merging**